core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -657,6 +681,13 @@ class BlockManagerMasterEndpoint(
     }
   }
 
+  private def getShufflePushMergerLocations(
+      numMergersNeeded: Int,
+      hostsToFilter: Set[String]): Seq[BlockManagerId] = {
+    val mergers = shuffleMergerLocations.values.filterNot(x => hostsToFilter.contains(x.host)).toSeq
+    mergers.take(numMergersNeeded)
```

Comment:
This is not prioritizing hosts with active executors right ?
Can we partition based on whether there are active executors or not, and use that to ensure we pick `numMergersNeeded` mergers (and fallback to hosts without mergers if not sufficient exists) ?

Reply:
This is not. I was holding off on that change as I thought we wanted to have more discussions on that front.

Reply:
I thought the reason we didn't do that internally is because the merger locations is picked at the beginning of the map stage and is only relevant with executor placement of the following reduce stage.
Since there is no guarantee on whether the active executors at the beginning of the map stage would still be active at the beginning of the reduce stage, we didn't do that.
Is that reasoning not applicable any more?